filament/backend/src/metal/MetalDriver.mm

```diff
@@ -824,6 +824,15 @@
         samplersToBind[binding] = samplerState;
     });
 
+    // Assign a default sampler to empty slots, in case Filament hasn't bound all samplers.
+    // Metal requires all samplers referenced in shaders to be bound.
+    for (auto& sampler : samplersToBind) {
+        if (!sampler) {
+            static SamplerParams defaultSampler;
```

Comment:
I wouldn't make it static here -- it's probably more costly than a regular stack object. In fact, I think you can just write `.getOrCreateState({})`

Reply:
Good call